chain/ethereum/src/chain.rs

```diff
@@ -68,50 +119,141 @@ impl Blockchain for Chain {
+        start_blocks: Vec<BlockNumber>,
+        filter: TriggerFilter,
+        metrics: Arc<BlockStreamMetrics>,
+    ) -> Result<BlockStream<Self>, Error> {
+        let logger = self
+            .logger_factory
+            .subgraph_logger(&deployment)
+            .new(o!("component" => "BlockStream"));
+        let chain_store = self.chain_store().clone();
+        let writable = self
+            .subgraph_store
+            .writable(&deployment)
+            .expect(&format!("no store for deployment `{}`", deployment.hash));
+        let chain_head_update_stream = self.chain_head_update_listener.subscribe(self.name.clone());
+
+        let requirements = filter.node_capabilities();
+
+        let triggers_adapter = self
+            .triggers_adapter(&deployment, &requirements)
+            .expect(&format!(
+                "no adapter for network {} with capabilities {}",
+                self.name, requirements
+            ));
+
+        Ok(BlockStream::new(
+            writable,
+            chain_store,
+            chain_head_update_stream,
+            triggers_adapter,
+            self.node_id.clone(),
+            deployment.hash,
+            filter,
+            start_blocks,
+            self.reorg_threshold,
+            logger,
+            metrics,
+            *MAX_BLOCK_RANGE_SIZE,
+            *TARGET_TRIGGERS_PER_BLOCK_RANGE,
+        ))
     }
 
     fn ingestor_adapter(&self) -> Arc<Self::IngestorAdapter> {
-        self.adapter.clone()
+        let eth_adapter = self.eth_adapters.cheapest().unwrap().clone();
+        let logger = self
+            .logger_factory
+            .component_logger(
+                "BlockIngestor",
+                Some(ComponentLoggerConfig {
+                    elastic: Some(ElasticComponentLoggerConfig {
+                        index: String::from("block-ingestor-logs"),
+                    }),
+                }),
+            )
+            .new(o!("provider" => eth_adapter.provider().to_string()));
+
+        let adapter = IngestorAdapter {
+            eth_adapter,
+            logger,
+            ancestor_count: self.ancestor_count,
+            chain_store: self.chain_store.clone(),
+        };
+        Arc::new(adapter)
+    }
+
+    fn node_capabilities(&self, archive: bool, traces: bool) -> Self::NodeCapabilities {
+        NodeCapabilities { archive, traces }
+    }
+
+    fn chain_store(&self) -> Arc<dyn ChainStore> {
+        self.chain_store.clone()
     }
 }
 
-pub struct DummyBlock;
+// ETHDEP: Wrapper until we can move BlockFinality into this crate
+pub struct WrappedBlockFinality(pub(crate) BlockFinality);
 
-impl Block for DummyBlock {
+impl Block for WrappedBlockFinality {
     fn ptr(&self) -> BlockPtr {
-        todo!()
+        self.0.ptr()
     }
 
     fn parent_ptr(&self) -> Option<BlockPtr> {
-        todo!()
+        self.0.parent_ptr()
     }
 }
 
-pub struct DummyDataSource;
+pub struct WrappedDataSource(DataSource);
```

Comment:
`WrappedDataSource` looks like dead code to me.

Reply:
Removed.